core/src/main/java/pl/agh/edu/management/advertisement/AdvertisementHandler.java

```diff
@@ -32,14 +39,31 @@ private boolean collide(AdvertisementCampaign campaign, LocalDate startDate, Loc
 
 	public void buyAdvertisementCampaign(AdvertisementType type, LocalDate startDate, LocalDate endDate) {
 		AdvertisementCampaign advertisementCampaign = new AdvertisementCampaign(JSONAdvertisementDataLoader.advertisementData.get(type), startDate, endDate);
-
+		bankAccountHandler.registerExpense(getCampaignFullCost(type, ChronoUnit.DAYS.between(startDate, endDate)));
 		timeCommandExecutor.addCommand(new TimeCommand(
 				() -> advertisementCampaigns.remove(advertisementCampaign),
 				endDate.atTime(LocalTime.MIDNIGHT).minusMinutes(1)));
 
 		advertisementCampaigns.add(advertisementCampaign);
 	}
 
+	public static BigDecimal getCampaignFullCost(AdvertisementType type, long noDays) {
+		return JSONAdvertisementDataLoader.advertisementData.get(type).costPerDay()
+				.multiply(BigDecimal.valueOf(noDays))
+				.multiply(BigDecimal.ONE.subtract(getDiscount(noDays)));
+	}
+
+	private static BigDecimal getDiscount(long noDays) {
```

Comment:
nie lepiej sobie takie rzeczy przechowywać w zewnetrzych plikach do pozniejszej konfiguracji, a jak już nie to bym chociaz zoptymalizował i wklepał odrazu wartości odjęte od 1 oraz zamienił te if'y na switch'a

Reply:
no nie wiem jak to na switcha zamienić. 
chat mi wypluł coś takiego:
    switch (true) {
        case (noDays >= 28):
            discount = new BigDecimal("0.2");
            break;
        case (noDays >= 14):
            discount = new BigDecimal("0.1");
            break;
        default:
            discount = BigDecimal.ZERO;
            break;
    }

Reply:
wg mnie trzymanie tego w jsonach to przerost formy nad treścią. Nikomu by się nie chciało tego zmieniać

Reply:
np.
```java
private static BigDecimal getDiscount(long noDays) {
  return switch (noDays) {
    case >= 28 -> new BigDecimal("0.2");
    case >= 14 -> new BigDecimal("0.1");
    default -> BigDecimal.ZERO;
  };
}
```
tamten potwór to nw co to jest

Reply:
![image](https://github.com/Hotel-Simulator/Hotel-Simulator/assets/80721230/9e5e2fbf-611d-4620-8d66-c8032688f56f)

Reply:
dobra za mocno wlazł kotlin bo obczajeniu sprawy bardziej szczegółowo jest opcja żeby to zrobić jakoś ładnie ale trzeba zrobić upgrade javy do najnowaszego LTS 21 czytaj podmienić liczby w build.gradle i wtedy możemy użyć konstrukcji
```java
	private static BigDecimal getDiscount(Long noDays) {
		return switch (noDays) {
			case Long value when value>28 -> new BigDecimal("0.2");
			case Long value when value>14 -> new BigDecimal("0.1");
			default -> BigDecimal.ZERO;
		};
	}
```
ale wybór zostawiam tobie